crates/gc/tests/all.rs

```diff
@@ -0,0 +1,146 @@
+extern crate parity_wasm;
+extern crate rayon;
+extern crate tempfile;
+extern crate wasm_bindgen_gc;
+
+use std::env;
+use std::error::Error;
+use std::fs;
+use std::io;
+use std::path::{Path, PathBuf};
+use std::process::{Command, Stdio};
+
+use rayon::prelude::*;
+use parity_wasm::elements::Module;
+use tempfile::NamedTempFile;
+
+struct Test {
+    input: PathBuf,
+}
+
+fn main() {
+    let mut tests = Vec::new();
+    find_tests(&mut tests, "tests/wat".as_ref());
+
+    run_tests(&tests);
+}
+
+fn find_tests(tests: &mut Vec<Test>, path: &Path) {
```

Comment:
Rather than implementing our own test harness, I think it is usually easier/better to have a build.rs generate normal `#[test]`s and let cargo be the harness. Then we wouldn't have to fiddle with rayon to get proper parallel test running, and we wouldn't need to create our own `Test` type and keep track of results ourselves, etc etc

The downside is that we have to create another test-only crate to avoid making a build.rs that isn't needed for normal builds

Reply:
Yeah I opted to not grow another crate because we already have so many, and I figured this logic wasn't really all that complicated and will probably mirror 90% of what will be in the build script anyway

Reply:
@fitzgen would you be ok leaving this as-is? I can remove some of the fanciness like rayon to simplify this to just executing tests which may make it much more amenable

Reply:
If we aren't going to generate `#[test]`s in `build.rs`, then we should use rayon.

Also, I should have mentioned this in the original review, but these tests are awesome, and it is great to see this PR! ^_^

Reply:
Ok! FWIW rayon lived up to its promise and was a 2 line change to parallelize, so it's easy enough to back out too!